crates/server/src/admin/handlers/blob.rs

```diff
@@ -173,36 +122,12 @@ pub async fn download_handler(
 
     // Get blob data
     match state.node_client.get_blob_bytes(&blob_id).await {
-        Ok(Some(blob_data)) => {
-            // Detect MIME type from content
-            let content_type = detect_mime_type(&blob_data);
-
-            // Generate a more descriptive filename based on content type
-            let filename = match content_type {
-                "image/png" => format!("blob-{}.png", &blob_id.to_string()[..8]),
-                "image/jpeg" => format!("blob-{}.jpg", &blob_id.to_string()[..8]),
-                "image/gif" => format!("blob-{}.gif", &blob_id.to_string()[..8]),
-                "image/webp" => format!("blob-{}.webp", &blob_id.to_string()[..8]),
-                "application/pdf" => format!("blob-{}.pdf", &blob_id.to_string()[..8]),
-                "application/zip" => format!("blob-{}.zip", &blob_id.to_string()[..8]),
-                "text/html" => format!("blob-{}.html", &blob_id.to_string()[..8]),
-                "application/json" => format!("blob-{}.json", &blob_id.to_string()[..8]),
-                "application/xml" => format!("blob-{}.xml", &blob_id.to_string()[..8]),
-                "text/plain" => format!("blob-{}.txt", &blob_id.to_string()[..8]),
-                _ => format!("blob-{}", &blob_id.to_string()[..8]),
-            };
-
-            Response::builder()
-                .status(StatusCode::OK)
-                .header(header::CONTENT_TYPE, content_type)
-                .header(header::CONTENT_LENGTH, blob_data.len())
-                .header(
-                    header::CONTENT_DISPOSITION,
-                    format!("attachment; filename=\"{}\"", filename),
-                )
-                .body(Body::from(blob_data.to_vec()))
-                .unwrap()
-        }
+        Ok(Some(blob_data)) => Response::builder()
+            .status(StatusCode::OK)
+            .header(header::CONTENT_TYPE, "application/octet-stream")
+            .header(header::CONTENT_LENGTH, blob_data.len())
+            .body(Body::from(blob_data.to_vec()))
```

Comment:
why not get_blob? we don't need the whole blob in memory

https://docs.rs/axum/latest/axum/body/struct.Body.html#method.from_stream